src/config/AppConfig.ts

```diff
@@ -1,37 +1,57 @@
 import "jsr:@std/dotenv/load";
 
-export const getEnvironmentVariable = (name: string): string | undefined =>
-	Deno.env.get(name);
-
 export interface AppConfig {
-	TelegramToken: string;
-	DbConnectionString: string;
-	DefaultLanguage: string;
-	IsDebug: boolean;
+	dbConnectionString: string;
+	telegramToken: string;
+	defaultLanguage: string;
+	isDebug: boolean;
 }
 
-export interface EnvVariable {
-	required: boolean;
-	default?: string;
-}
+export const prepareAppConfig = (): AppConfig => {
+	const required = getRequiredVars();
+	const optional = getOptionalVars();
 
-export const AppConfigEnvVariables = {
-	TELEGRAM_TOKEN: {
-		required: true,
-		value: getEnvironmentVariable("TELEGRAM_TOKEN"),
-	},
-	DB_CONNECTION_STRING: {
-		required: true,
-		value: getEnvironmentVariable("DB_CONNECTION_STRING"),
-	},
-	DEFAULT_LANGUAGE: {
-		required: false,
-		default: "en",
-		value: getEnvironmentVariable("DEFAULT_LANGUAGE"),
-	},
-	DEBUG: {
-		required: false,
-		default: "0",
-		value: getEnvironmentVariable("DEBUG"),
-	},
+	return {
+		dbConnectionString: required.DB_CONNECTION_STRING,
+		telegramToken: required.TELEGRAM_TOKEN,
+		defaultLanguage: optional.DEFAULT_LANGUAGE,
+		isDebug: optional.DEBUG === "1",
+	};
 };
+
+export function getRequiredVars() {
+	const requiredVars = [
+		"TELEGRAM_TOKEN",
+		"DB_CONNECTION_STRING",
+	] as const;
+
+	const res: { [key: string]: string } = {};
+	for (const key of requiredVars) {
+		const env = Deno.env.get(key);
+		if (!env) throw new Error(`${key} is required for project startup.`);
+
+		res[key] = env;
+	}
+
+	return res as {
+		[key in (typeof requiredVars)[number]]: string;
+	};
```

Comment:
The `(typeof requiredVars)[number]` is used to convert an array into a type union of its items' types. The `[number]` part is an ["Indexed Access Types"][1] syntax.

[1]: https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html